framework/audio/engine/iaudiodriver.h

```diff
@@ -0,0 +1,69 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IAUDIODRIVER_H
+#define MU_AUDIO_IAUDIODRIVER_H
+
+#include <cstdint>
+#include <string>
+#include <functional>
+
+#include "modularity/imoduleexport.h"
+
+namespace mu {
+namespace audio {
+namespace engine {
+
+class IAudioDriver : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(IAudioDriver)
+
+public:
+    virtual ~IAudioDriver() = default;
+
+    using DeviceID = uint32_t;
+
+    enum class Format {
+        AudioF32,
```

Comment:
What is an F32 and S16?
I think a comment will be useful

Reply:
These are audio formats, for those who are even a little familiar with audio, this is a well-known thing. There are many formats themselves, not only these two. I’m exploring, it may be better to choose one and all.

Reply:
added comments